### ISSUE TITLE
[Radio][RadioGroup] missing aria-required

### ISSUE THREAD

**Issue by cgatian** [mui/base-ui#2226]
# Bug report

## Current behavior

`RadioGroup` accepts a `required` prop, but doesn't use it on the element setting the `role=radiogroup`.

## Expected behavior

When `required` is passed, `aria-required` should be added to the element with `role=radiogroup`

> The required state is specified on the radiogroup element rather than one of the radio elements.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/radiogroup_role#associated_wai-aria_roles_states_and_properties

Code change here:

https://github.com/mui/base-ui/blob/1b273e8ad224451afb573a88769c1bd57ef8ed46/packages/react/src/radio-group/RadioGroup.tsx#L205-L208

## Base UI version

1.0.0-beta.0
